### PR TITLE
Ensure session is only fetched after lock is held

### PR DESCRIPTION
**Pull request by cmyui**

### CODE REVIEW

File: handlers/mainHandler.py [hunk 1]
```diff
@@ -160,23 +161,33 @@ def asyncPost(self) -> None:
...
         else:
+            # Make sure token is valid syntax
+            try:
+                UUID(requestTokenString)
+            except ValueError:
+                # Invalid token, ignore request
+                self.set_status(400)
+                self.add_header("Content-Type", "text/html; charset=UTF-8")
+                self.write(HTML_PAGE)
+                return
```
Comment (cmyui) [inline handlers/mainHandler.py]: prevents people from injecting funny data into redis

**Review by tsunyoku — DISMISSED** [review]

File: handlers/mainHandler.py [hunk 2]
```diff
@@ -214,20 +225,18 @@ def asyncPost(self) -> None:
                     "Server has restarted.",
                 ) + serverPackets.banchoRestart(0)
             finally:
-                userToken = osuToken.get_token(requestTokenString)
```
Comment (cmyui) [inline handlers/mainHandler.py:217]: i have no idea why this was being fetch again here?

Comment (cmyui) [reply handlers/mainHandler.py:217]: hmm actually perhaps this is because `bancho_packets[packetID](userToken, packetData)` can modify it

Comment (tsunyoku) [reply handlers/mainHandler.py:217]: is it in-case one of the packet handlers modifies the session?

Comment (cmyui) [reply handlers/mainHandler.py:217]: re-added

Comment (tsunyoku) [reply handlers/mainHandler.py:217]: ~~thinking for example L228, a packet may set kicked but we don't have the updated session..?~~ yep

**Comment by cmyui** [comment]
seems to do the trick

**Pull request merged**